snuba/replacers/replacements_and_expiry.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+import logging
+from datetime import datetime, timedelta
+from typing import Mapping, Sequence
+
+logger = logging.getLogger(__name__)
+
+
+from snuba.redis import RedisClientKey, get_redis_client
+from snuba.state import get_int_config
+
+redis_client = get_redis_client(RedisClientKey.REPLACEMENTS_STORE)
+config_auto_replacements_bypass_projects_hash = (
+    "snuba-config-auto-replacements-bypass-projects-hash"
+)
+
+REPLACEMENTS_EXPIRY_WINDOW_MINUTES_KEY = "replacements_expiry_window_minutes"
+
+
+def get_minutes(key: str, default: int) -> int:
```

Comment:
> this function is not descriptively named. What are the minutes for?

We can either get the expiry minutes (ie project should be skipped for 5 minutes) or the counter window (which is the time window we use to calculate the threshold for determining if we should skip a project). I renamed it to `get_expiry_window_or_counter_window_size`. The main purpose of this function, though, is to get around the type checking, since in our case the default value (5 or 10) should be returned instead of `None` anyways.

> Also is there ever a time you would have the arguments be anything other than:
`get_minutes(REPLACEMENTS_EXPIRY_WINDOW_MINUTES_KEY, 5)`

Yes: https://github.com/getsentry/snuba/pull/5883/files#diff-44863b6f784105daa8ce60ea6f4c8db7cb46e122dec3e3c0c59d8ec3d96b1fbaR30